### ISSUE TITLE
Extend the kexec image with more packages and firmware

### ISSUE THREAD

**Issue by wokalski** [nix-community/nixos-images#291]
Hey, i spent a bunch of time on thursday debugging my provisioning and it turned out some firmware for my NICs was only accessible with `hardware.enableAllFirmware = true;`.

I am happy to have solved the issue (Intel E810 NICs in my case) but for the sake of future travelers I would personally be in favor of including this option in the default kexec image. Would save some headache. Additionally, I would also probably be in favor of including:
1. ethtool
2. lshw
3. pciutils
4. tcpdump
5. dmidecode
6. udev
7. hwinfo

just for debugging

**Comment by Mic92** [comment]
We need to keep the non-interactive kexec we have a tight memory budget and we are already doing some crazy space optimization to keep the memory usage low to be able to support 1GB VMs. Having more firmware sounds ok.

However I am not so sure, I want to maintain this non-interactive image in future for much longer as I only need the kexec non-interactive image for nixos-anywhere.
However are you aware of the kexec image in nixos? It's accessible through nixos-generators: https://github.com/nix-community/nixos-generators
This allows you to put in arbitrary nixos configuration.

**Comment by wokalski** [comment]
I was not aware of it. I was using this image solely for nixos anywhere but I'm happy to close this is nixos anywhere will make it possible to use an arbitrary config with the generators.

**Comment by thamenato** [comment]
I had a somewhat similar situation here in regards to not being able to access my host after `kexec` starts (I am using `nixos-anywhere`) due to some very specific networking configuration that uses `bonding`. The way I ended up fixing it was to extend the `non-interactive` from this repository and adding the changes I needed (like enabling the "bonding" kernel module).

Here's my `flake.nix` - 

```nix
{
  description = "Custom Kexec Images";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable-small";
    # get nixos modules from the nixos-images repository
    nixos-images.url = "github:nix-community/nixos-images";
  };

  outputs =
    {
      self,
      nixpkgs,
      nixos-images,
    }:
    let
      system = "x86_64-linux";
      kexec-installer =
        module:
        (nixpkgs.legacyPackages.${system}.nixos [
          module
          nixos-images.nixosModules.kexec-installer
        ]).config.system.build.kexecTarball;

    in
    {
      packages.${system} = {
        pnap-kexec =
          let
            module = {
              system.kexec-installer.name = "pnap-kexec";
              imports = [
                nixos-images.nixosModules.noninteractive
                # my nix file config goes here
                ./custom.nix
              ];
            };
          in
          kexec-installer module;
      };
    };
}

```

**Comment by ben-auo** [comment]
I agree with keeping the image lean, but is there no editor? (I tried nano, vim, and vi...)

**Comment by Mic92** [comment]
There is an interactive kexec image that does have an editor. The non-interactive image is for nixos-anywhere where you don't need to edit any files. You can however specify the interactive kexec image if you want and have enough RAM.

**Comment by Mic92** [comment]
Also we can maybe enable all firmware.

**Comment by tcurdt** [comment]
I am trying to find a way to build a custom image (a little pre-configured e.g. with ssh keys) that I can upload to Hetzner (as basis for cloud machines). 

I got an ISO working. And [I tried to build a disk image manually](https://github.com/tcurdt/nixos-unattended-install-iso/actions/runs/13253535774/job/36996202948#step:4:2098) - but I am well beyond my comfort zone here.

Question is: Could I base my flake off this image somehow maybe?
I would appreciate any pointers.
